File: app/src/main/java/com/quranapp/android/utils/chapterInfo/ChapterInfoJSInterface.java
package com.quranapp.android.utils.chapterInfo;

import android.webkit.JavascriptInterface;
import android.widget.Toast;

import com.quranapp.android.activities.ActivityChapInfo;
import com.quranapp.android.components.quran.QuranMeta;
import com.quranapp.android.utils.Log;
import com.quranapp.android.utils.univ.MessageUtils;

public class ChapterInfoJSInterface {
    private final ActivityChapInfo mActivity;

    public ChapterInfoJSInterface(ActivityChapInfo activityChapInfo) {
        mActivity = activityChapInfo;
    }

    @JavascriptInterface
    public void openReference(int chapterNo, int fromVerse, int toVerse) {
        QuranMeta quranMeta = mActivity.mQuranMeta;

        if (!QuranMeta.isChapterValid(chapterNo) || !quranMeta.isVerseRangeValid4Chapter(chapterNo, fromVerse,
            toVerse)) {
            Log.d(chapterNo, fromVerse, toVerse);
            MessageUtils.INSTANCE.showRemovableToast(mActivity, "Could not open references", Toast.LENGTH_LONG);
            return;
        }

        mActivity.showChapterQuickReference(chapterNo, fromVerse, toVerse);
    }
}

File: app/src/main/java/com/quranapp/android/utils/chapterInfo/ChapterInfoJSInterface.kt
package com.quranapp.android.utils.chapterInfo

import android.content.Context
import android.webkit.JavascriptInterface
import android.widget.Toast
import com.quranapp.android.utils.quran.QuranMeta
import com.quranapp.android.utils.univ.MessageUtils

class ChapterInfoJSInterface(
    private val context: Context,
    private val verseCount: Int,
    private val onOpenReference: (chapterNo: Int, fromVerse: Int, toVerse: Int) -> Unit,
) {
    @JavascriptInterface
    fun openReference(chapterNo: Int, fromVerse: Int, toVerse: Int) {
        if (!QuranMeta.isChapterValid(chapterNo)
            || fromVerse < 1 || toVerse < 1
            || fromVerse > toVerse
            || fromVerse > verseCount
            || toVerse > verseCount
        ) {
            MessageUtils.showRemovableToast(context, "Could not open references", Toast.LENGTH_LONG)
            return
        }

        onOpenReference(chapterNo, fromVerse, toVerse)
    }
}
